sunpy/image/transform.py

```diff
@@ -167,7 +101,176 @@ def _get_transform_method(method, use_scipy):
         try:
             import skimage  # NoQA
         except ImportError:
-            warn_user("scikit-image could not be imported. Image rotation will use scipy.")
-            method = 'scipy'
+            raise ImportError("scikit-image must be installed to be usable for rotation.")
 
     return method
+
+
+def add_rotation_function(name, *, handles_clipping=False,
+                          handles_image_nans=False, handles_nan_missing=False):
+    """
+    Decorator to add a rotation function to the library of selectable
+    implementations.
+
+    The decorator accepts the parameters listed under ``Parameters``.  The decorated
+    function must accept the parameters listed under ``Other Parameters`` in that
+    order and return the rotated image.
+
+    Parameters
+    ----------
+    name : `str`
+        The name that will be used to select the rotation function
+    handles_clipping : `bool`
+        Specifies whether the rotation function can internally perform clipping
+    handles_image_nans : `bool`
+        Specifies whether the rotation function can internally handle NaNs in the
+        input image
+    handles_nan_missing : `bool`
+        Specifies whether the rotation function can internally handle NaN as the
+        missing value
+
+    Other Parameters
```

Comment:
I think we could use some documentation on the output requirements, i.e. clipping, NaNs etc with reference to the fact that this decorator can handle those things for you? i.e. If I want to write a new function when should I use the keyword args to the decorator, what if they slow my implementation down etc.

Reply:
Yeah, I can add more description.  In particular, the additional convolution if we have to externally handle image NaNs should be noted.

Reply:
Some more description added, but perhaps you still want more

Reply:
It still feels like it needs some kind of summary for someone coming at this cold. This docstring covers all the technical details but if we threw it at someone who doesn't have any context I don't think they would be able to parse this and write a function and be confident they have handled everything correctly.

Perhaps the best place for this isn't here it's in the dev guide?